src/app/codegen/code_generator.py

```diff
@@ -49,13 +53,25 @@ def generate(template: GenerationTemplate, welding_points: List[WeldingPoint]) -
         # Define a dictionary with functions, which can be used in templates
         func_dict = {"generate_blockly_block_id": TemplateFunctions.generate_blockly_block_id}
 
+        # Robot for which the program will be generated
+        robot = welding_points[0].robot
+        if robot is None:
+            raise CodeGeneratorException("Welding point does not have a robot")
```

Comment:
```suggestion
        robots = set([wp.robot.id for wp in welding_points])
        if len(robots) !== 1 or robots[0] is None:
            raise CodeGeneratorException("All welding points must have the same non-None robot")
        robot = welding_points[0].robot
```
I think we should rather validate whether all the robots have the same non-None robot. (Although "non-None" does sound awful)

Reply:
From application perspective, it's also not possible that this method gets called with a `None` robot type (except for the unit test), but stuff like this should still be checked. Thanks for the catch!